awesome_owl/static/src/counter/counter.js

```diff
@@ -0,0 +1,19 @@
+/** @odoo-module **/
+
+import { Component, useState } from "@odoo/owl";
+
+export class Counter extends Component {
+    static template = "awesome_owl.counter";
+    state = useState({ value: 0 });
+
+    static props = {
+        onChange: { type: Function, optional: true },
+    };
+
+    increment() {
+        this.state.value++;
+        if (this.props.onChange) {
+            this.props.onChange(this.state.value);
+        }
```

Comment:
tip: equivalent syntactic sugar
```suggestion
        this.props.onChange?.()

```